indexing-service/src/main/java/org/apache/druid/indexing/overlord/http/OverlordCompactionResource.java

```diff
@@ -115,13 +116,11 @@ public Response getCompactionSnapshots(
     } else {
       AutoCompactionSnapshot autoCompactionSnapshot = scheduler.getCompactionSnapshot(dataSource);
       if (autoCompactionSnapshot == null) {
-        return Response.status(Response.Status.NOT_FOUND)
-                       .entity(Collections.singletonMap("error", "Unknown DataSource"))
-                       .build();
+        return ServletResourceUtils.buildErrorResponseFrom(NotFound.exception("Unknown DataSource"));
       }
       snapshots = Collections.singleton(autoCompactionSnapshot);
     }
-    return Response.ok(Collections.singletonMap("latestStatus", snapshots)).build();
+    return Response.ok(new CompactionStatusResponse(snapshots)).build();
```

Comment:
Is there a reason that raw map isn't compatible but a class works? I believe both would be serialised in the same manner.

Reply:
Yes, that's true. The problem wasn't that I had used a map instead of a dedicated POJO. Instead, I had forgotten to use anything at all 😛 .

The Overlord REST endpoint returns a response that looks like `{"latestStatus": [{snapshot1object}, {snapshot2object}, {snapshot3object}]}` but in the Overlord client, I was just reading the internal list `[{snapshot1object}, {snapshot2object}, {snapshot3object}]` rather than the whole object.

Reply:
to avoid such mistakes in the future and to keep the code simple, I decided to use a model POJO instead of maps everywhere.

Reply:
Agree. Makes it so much cleaner.